frontend/packages/process-editor/src/components/ConfigPanel/ConfigContent/EditDataType/SelectDataType.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { NativeSelect } from '@digdir/design-system-react';
+import { StudioButton, StudioDeleteButton } from '@studio/components';
+import type { MetaDataForm } from '../../../../contexts/BpmnConfigPanelContext';
+import { useBpmnConfigPanelFormContext } from '../../../../contexts/BpmnConfigPanelContext';
+import { useBpmnApiContext } from '../../../../contexts/BpmnApiContext';
+import { useTranslation } from 'react-i18next';
+import { XMarkIcon } from '@studio/icons';
+import classes from './SelectDataType.module.css';
+
+export interface SelectDataTypeProps {
+  dataModelIds: string[];
+  existingDataType: string;
+  connectedTaskId: string;
+  onClose: () => void;
+}
+export const SelectDataType = ({
+  dataModelIds,
+  existingDataType,
+  connectedTaskId,
+  onClose,
+}: SelectDataTypeProps) => {
+  const { t } = useTranslation();
+  const { updateDataType } = useBpmnApiContext();
+  const { metaDataFormRef } = useBpmnConfigPanelFormContext();
+  const handleChangeDataModel = (dataModelId: string) => {
+    if (dataModelId === existingDataType) return;
+    const newMetadata: MetaDataForm = {
+      dataTypeChangeDetails: {
+        newDataType: dataModelId,
+        connectedTaskId: connectedTaskId,
+      },
+    };
+    metaDataFormRef.current = Object.assign({}, metaDataFormRef.current, newMetadata);
+    updateDataType(metaDataFormRef.current);
+  };
+
+  return (
+    <div className={classes.dataTypeSelect}>
+      <NativeSelect
+        size='small'
+        onChange={({ target }) => {
```

Comment:
Maybe we could refactor the code in the onChange out in its own function ```handleChange``` ? 😄 Increases readability